taskcluster/fenix_taskgraph/parameters.py

```diff
@@ -7,49 +7,49 @@
 import os
 import re
 
+from mozilla_version.fenix import FenixVersion
 from six import text_type
 from taskgraph.parameters import extend_parameters_schema
 from voluptuous import All, Any, Optional, Range, Required
 
-
-BETA_SEMVER = re.compile(r'^v\d+\.\d+\.\d+-beta\.\d+$')
-RELEASE_SEMVER = re.compile(r'^v\d+\.\d+\.\d+(-rc\.\d+)?$')
-
+from .release_promotion import read_version_file
 
 extend_parameters_schema({
     Required("pull_request_number"): Any(All(int, Range(min=1)), None),
     Required("release_type"): text_type,
     Optional("shipping_phase"): Any('build', 'ship', None),
     Required("version"): text_type,
+    Required("next_version"): Any(None, text_type),
 })
 
 
 def get_decision_parameters(graph_config, parameters):
+    parameters.setdefault("release_type", "")
     head_tag = parameters["head_tag"].decode("utf-8")
-    parameters["release_type"] = resolve_release_type(head_tag)
     parameters["version"] = head_tag[1:] if head_tag else ""
 
     pr_number = os.environ.get("MOBILE_PULL_REQUEST_NUMBER", None)
     parameters["pull_request_number"] = None if pr_number is None else int(pr_number)
+    parameters.setdefault("next_version", None)
 
+    # TODO: Remove this block once github-releases are not supported anymore
     if parameters["tasks_for"] == "github-release":
-        for param_name in ("release_type", "version"):
-            if not parameters[param_name]:
-                raise ValueError(
-                    'Cannot run github-release if "{}" is not defined. Got: {}'.format(
-                        param_name, parameters[param_name]
-                    )
-                )
         parameters["target_tasks_method"] = "release"
 
-
-def resolve_release_type(head_tag):
-    if not head_tag:
-        return ""
-    elif BETA_SEMVER.match(head_tag):
-        return "beta"
-    elif RELEASE_SEMVER.match(head_tag):
-        return "release"
-    else:
-        raise ValueError('Github tag must be in semver format and prefixed with a "v", '
-                         'e.g.: "v1.0.0-beta.0" (beta), "v1.0.0-rc.0" (release) or "v1.0.0" (release)')
+        version_string = parameters["version"]
+        version_in_file = read_version_file()
+        if version_string != version_in_file:
+            raise ValueError("Version given in tag ({}) does not match the one in version.txt ({})".format(version_string, version_in_file))
```

Comment:
We want to allow both shipit and github-releases for some time, while people get used to the new procedure. Since shipit only relies on `version.txt` to guess version numbers, we have to make sure there is no mismatch when someone creates a github-release. We actually have the same logic on android-components https://github.com/mozilla-mobile/android-components/blob/d61ad31b366c855a6d7578f392c0c8972712d3a9/taskcluster/ac_taskgraph/__init__.py#L58

This error has been tested in staging:

```
[task 2020-11-12T16:59:13.085Z] 2020-11-12 16:59:13,085 - WARNING - using default project parameters; add fenix to PER_PROJECT_PARAMETERS in /builds/worker/.local/lib/python2.7/site-packages/taskgraph/decision.pyc to customize behavior for this project
[task 2020-11-12T16:59:13.086Z] Traceback (most recent call last):
[task 2020-11-12T16:59:13.086Z]   File "/builds/worker/.local/lib/python2.7/site-packages/taskgraph/main.py", line 444, in main
[task 2020-11-12T16:59:13.087Z]     args.command(vars(args))
[task 2020-11-12T16:59:13.087Z]   File "/builds/worker/.local/lib/python2.7/site-packages/taskgraph/main.py", line 321, in decision
[task 2020-11-12T16:59:13.087Z]     taskgraph_decision(options)
[task 2020-11-12T16:59:13.087Z]   File "/builds/worker/.local/lib/python2.7/site-packages/taskgraph/decision.py", line 94, in taskgraph_decision
[task 2020-11-12T16:59:13.087Z]     write_artifact('parameters.yml', dict(**tgg.parameters))
[task 2020-11-12T16:59:13.087Z]   File "/builds/worker/.local/lib/python2.7/site-packages/taskgraph/generator.py", line 140, in parameters
[task 2020-11-12T16:59:13.088Z]     return self._run_until('parameters')
[task 2020-11-12T16:59:13.088Z]   File "/builds/worker/.local/lib/python2.7/site-packages/taskgraph/generator.py", line 353, in _run_until
[task 2020-11-12T16:59:13.088Z]     k, v = self._run.next()
[task 2020-11-12T16:59:13.088Z]   File "/builds/worker/.local/lib/python2.7/site-packages/taskgraph/generator.py", line 236, in _run
[task 2020-11-12T16:59:13.089Z]     parameters = self._parameters(graph_config)
[task 2020-11-12T16:59:13.089Z]   File "/builds/worker/.local/lib/python2.7/site-packages/taskgraph/decision.py", line 80, in <lambda>
[task 2020-11-12T16:59:13.089Z]     lambda graph_config: get_decision_parameters(graph_config, options)
[task 2020-11-12T16:59:13.089Z]   File "/builds/worker/.local/lib/python2.7/site-packages/taskgraph/decision.py", line 197, in get_decision_parameters
[task 2020-11-12T16:59:13.089Z]     find_object(graph_config['taskgraph']['decision-parameters'])(graph_config, parameters)
[task 2020-11-12T16:59:13.089Z]   File "taskcluster/fenix_taskgraph/parameters.py", line 42, in get_decision_parameters
[task 2020-11-12T16:59:13.089Z]     raise ValueError("Version given in tag ({}) does not match the one in version.txt ({})".format(version_string, version_in_file))
[task 2020-11-12T16:59:13.090Z] ValueError: Version given in tag (82.1.99) does not match the one in version.txt (82.1.8)
```

https://firefox-ci-tc.services.mozilla.com/tasks/AuxX3sTMSmWnVjduyvqSLA/runs/0/logs/https%3A%2F%2Ffirefox-ci-tc.services.mozilla.com%2Fapi%2Fqueue%2Fv1%2Ftask%2FAuxX3sTMSmWnVjduyvqSLA%2Fruns%2F0%2Fartifacts%2Fpublic%2Flogs%2Flive.log

Reply:
Later edit: nevermind the above. Regardless of it being triggered from Ship-it or via `github-release`, the task group that we generate, both contain the `version-bump` tasks so we can't fall behind.